omc3/scripts/linfile_clean.py

```diff
@@ -0,0 +1,234 @@
+If the ``backup`` option is activated, a backup of the original file
+is created, which can be restored via the ``restore`` option.
+In case the restore-flag is given, only the filenames are required.
+No cleaning is performed with this option.
+
+**Arguments:**
+
+*--Required--*
+
+- **files** *(PathOrStr)*:
+
+    List of paths to the lin-files, including suffix.
+
+
+*--Optional--*
+
+- **backup**:
+
+    Create a backup of the files before overwriting. The backups are
+    numbered, with the highest number being the latest backup.
+
+    action: ``store_true``
+
+
+- **columns** *(str)*:
+
+    Columns to clean on.
+
+
+- **limit** *(float)*:
+
+    Do not clean data-points deviating less than this limit from the
+    average.
+
+    default: ``0.0``
+
+
+- **restore**:
+
+    Restore the latest backup of the files. If this parameter is given, no
+    cleaning is performed.
+
+    action: ``store_true``
+
+
+TODO:
+also use isolation forest, BUT this probably needs some modification there
+as well, as it only cleans on TUNE, not on NATTUNE.
+And it requires an accelerator instance.
+"""
+import shutil
+from numbers import Number
+from pathlib import Path
+from typing import Sequence, Union
+
+import pandas as pd
+import tfs
+from generic_parser.entrypoint_parser import (
+    entrypoint, EntryPointParameters
+)
+
+from omc3.definitions.formats import BACKUP_FILENAME
+from omc3.harpy.constants import COL_NAME
+from omc3.utils import logging_tools
+from omc3.utils.iotools import PathOrStr, save_config
+from omc3.utils.outliers import get_filter_mask
+
+LOG = logging_tools.get_logger(__name__)
+
+
+def get_params():
+    return EntryPointParameters(
+        files=dict(
+            required=True,
+            type=PathOrStr,
+            nargs='+',
+            help="List of paths to the lin-files, including suffix.",
+        ),
+        # restore backup
+        restore=dict(
+            action="store_true",
+            help=("Restore the latest backup of the files. "
+                  "If this parameter is given, no cleaning is performed."),
+        ),
+        # for actual cleaning
+        columns=dict(
+            nargs='+',
+            type=str,
+            help="Columns to clean on.",
+        ),
+        limit=dict(
+            type=float,
+            help="Do not clean data-points deviating less than this limit from the average.",
+            default=0.0,
+        ),
+        backup=dict(
+            action="store_true",
+            help=("Create a backup of the files before overwriting. "
+                  "The backups are numbered, with the highest number being "
+                  "the latest backup.")
+        ),
+    )
+
+
+@entrypoint(get_params(), strict=True)
+def main(opt):
+    """Main function, to parse commandline input and separate restoration
+    from cleaning."""
+    save_config(Path('.'), opt, __file__)
+    if opt.restore:
+        restore_files(opt.files)
+        return
```

Comment:
if you use this script as a function, it might be nice to return the files paths or something

Reply:
yeah. I thought about that, but then I realized, that the filepaths you return are just the file paths you put in...

Reply:
except for the files which couldnt be found/backup(p?)ed

Reply:
but that throws an error. so nothing is returned anyway